lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -529,29 +558,38 @@ def get_students_opened_subsection(request, csv=False):
     student names, usernames for CSV download.
     """
     module_state_key = Location.from_deprecated_string(request.GET.get('module_id'))
+    course_id = request.GET.get('course_id')
     csv = request.GET.get('csv')
 
     # Query for "opened a subsection" students
     students = models.StudentModule.objects.select_related('student').filter(
         module_state_key__exact=module_state_key,
         module_type__exact='sequential',
-    ).values('student__username', 'student__profile__name').order_by('student__profile__name')
+    ).values('student__id', 'student__username', 'student__profile__name').order_by('student__profile__name')
+
+    course_key = locator.CourseLocator.from_string(course_id)
+    non_student_id_list = get_non_student_list(course_key)  # course_key is not a string
+    num_non_student = len(non_student_id_list)
 
     results = []
+    non_student_skipped = 0
     if not csv:
         # Restrict screen list length
-        # Adding 1 so can tell if list is larger than MAX_SCREEN_LIST_LENGTH
+        # Adding 1 + num_non_student so can tell if list is larger than MAX_SCREEN_LIST_LENGTH. non-students will be skipped.
         # without doing another select.
-        for student in students[0:MAX_SCREEN_LIST_LENGTH + 1]:
-            results.append({
-                'name': student['student__profile__name'],
-                'username': student['student__username'],
-            })
-
+        for student in students[0:MAX_SCREEN_LIST_LENGTH + 1 + num_non_student]:
```

Comment:
This looks really messy and is quite difficult for me to follow. I've not thought it through but is it possible to filter out the non-students in the query above and so simplify?